src/common/io/tls/client.c

```diff
@@ -351,7 +307,92 @@ tlsClientOpen(TlsClient *this)
         }
     }
 
-    FUNCTION_LOG_RETURN(TLS_SESSION, result);
+    FUNCTION_LOG_RETURN(IO_SESSION, result);
+}
+
+/**********************************************************************************************************************************/
+static const IoClientInterface tlsClientInterface =
+{
+    .type = &IO_CLIENT_TLS_TYPE_STR,
+    .open = tlsClientOpen,
+    .toLog = tlsClientToLog,
+};
+
+IoClient *
+tlsClientNew(SocketClient *socket, TimeMSec timeout, bool verifyPeer, const String *caFile, const String *caPath)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug)
+        FUNCTION_LOG_PARAM(SOCKET_CLIENT, socket);
+        FUNCTION_LOG_PARAM(TIME_MSEC, timeout);
+        FUNCTION_LOG_PARAM(BOOL, verifyPeer);
+        FUNCTION_LOG_PARAM(STRING, caFile);
+        FUNCTION_LOG_PARAM(STRING, caPath);
+    FUNCTION_LOG_END();
+
+    ASSERT(socket != NULL);
+
+    IoClient *this = NULL;
```

Comment:
wouldn't you normally declare this later, since you don't use it until later..?  Seems like it's here just because it was being used earlier before.

Reply:
It needs to be here because it must be outside the MEM_CONTEXT_NEW block and this is last chance to do that.